packages/maskbook/src/plugins/Wallet/services/keystore.ts

```diff
@@ -0,0 +1,144 @@
+import scrypt from 'scrypt-js'
+import Web3Utils from 'web3-utils'
+import { Buffer } from 'buffer'
+
+export type KeyStore = KeyStore.Type
+
+declare namespace KeyStore {
+    type Type = AESCipher & KeyDerivation & { mac: string }
+
+    interface AESCipher {
+        cipher: 'aes-128-ctr' | 'aes-128-cbc'
+        cipherparams: AESCipherParams
+        ciphertext: string
+    }
+
+    interface AESCipherParams {
+        iv: string
+    }
+
+    type KeyDerivation = PBKDF2 | Scrypt
+
+    interface PBKDF2 {
+        kdf: 'pbkdf2'
+        kdfparams: PBKDF2Params
+    }
+
+    interface PBKDF2Params {
+        c: number
+        prf: 'hmac-sha256'
+        dklen: number
+        salt: string
+    }
+
+    interface Scrypt {
+        kdf: 'scrypt'
+        kdfparams: ScryptParams
+    }
+
+    interface ScryptParams {
+        n: number
+        p: number
+        r: number
+        dklen: number
+        salt: string
+    }
+}
+
+export interface V3Keystore {
+    crypto: KeyStore
+    id: string
+    version: number
+    address: string
+}
+
+export async function fromV3Keystore(input: string | V3Keystore, password: string) {
+    const json: V3Keystore = typeof input === 'object' ? input : (JSON.parse(input) as V3Keystore)
+
+    if (json.version !== 3) {
+        throw new Error('Not a V3 wallet')
+    }
+
+    let derivedKey: Uint8Array, kdfparams: KeyStore.ScryptParams | KeyStore.PBKDF2Params
```

Comment:
```suggestion
    let derivedKey: Uint8Array
    let kdfparams: KeyStore.ScryptParams | KeyStore.PBKDF2Params
```

split variable definitions.